### DISCUSSION TITLE
Enable using RLS without the usage of built-in Supabase Auth

### DISCUSSION THREAD

**Discussion by jcheese1** [supabase/supabase#2074] [Feature Requests]
Hey,

So I currently use Prisma with Supabase, and I do all the authorization via just a simple Session table that keeps track of logged in users. It would be great if I can implement RLS with a custom auth solution, like in my case.

I'm thinking of something like this:

- Getting the user via session ID
- Check if the bucket they are trying to access is their own
- Depending on that allow read/write

on the side of Supabase.

I guess I can just have some sort of a middleware to check the above, but I think its safer to have it on the DB side.

I hope this makes sense!

**Comment by easy-street** [comment]
the approach described here should cover this use case: https://github.com/supabase/supabase/discussions/1849#discussioncomment-837636

essentially:
- mint custom JWTs with the authenticated user ID as a claim
- implement RLS policies with helper functions that extract values from the JWT

**Comment by jcheese1** [comment]
@easy-street interesting. How would I pass JWT on a per call basis though?

**Reply by easy-street** [reply]
depending on the specifics of your app, generally one of:
- create the supabase client with an `Authorization` header passed via options as per the example in the [linked comment](https://github.com/supabase/supabase/discussions/1849#discussioncomment-837636)
- utilize the (not yet documented) [`setAuth` method](https://github.com/supabase/supabase/issues/1818)
